uptrain/core/classes/monitors/data_integrity.py

```diff
@@ -14,17 +15,55 @@ def base_init(self, fw, check):
         self.threshold = check.get("threshold", None)
         self.count = 0
         self.num_issues = 0
-
+            
     def base_check(self, inputs, outputs, gts=None, extra_args={}):
+        # Perform measurable compute and log only if the measurable feature is
+        # present in the inputs
+        if self.measurable.col_name() not in inputs.keys():
+            return
         signal_value = self.measurable.compute_and_log(
             inputs, outputs, gts=gts, extra=extra_args
         )
+        outliers = None
         if self.integrity_type == "non_null":
             has_issue = signal_value == None
         elif self.integrity_type == "less_than":
             has_issue = signal_value > self.threshold
         elif self.integrity_type == "greater_than":
             has_issue = signal_value < self.threshold
+        elif self.integrity_type == "z_score":
+            if self.threshold is None:
+                self.threshold = 3
```

Comment:
Is it best to assign `self.threshold` to an arbitrary value like 3 or instead raise an exception to make user explicitly specify a threshold based on data?

Reply:
3 is not arbitrary; it's a good threshold from what I have read.